### ISSUE TITLE
Documentation for Custom resource format loaders is wrong

### ISSUE THREAD

**Issue by YeldhamDev** [godotengine/godot-docs#2411]
From @Darkheron:
> While writing a custom resource loader and following the directions here:
> https://docs.godotengine.org/en/3.1/development/cpp/custom_resource_format_loaders.html
> I have found that this documentation is very wrong and will certainly result in sigfaults/frustration :)
> 
> In particular, the documentation regarding the registration and unregistration is flawed.
> New loaders/savers need to be registered as such:
> 
> ```
> static Ref<MyLoaderText> my_loader_text; 
> void register_my_types()
> {
>     my_loader_text.instance();
>     ResourceLoader::add_resource_format_loader(m_loader_text);
> }
> ```
> 
> and unregistered as such:
> 
> ```
> void unregister_my_types()
> {
>     ResourceLoader::remove_resource_format_loader(my_loader_text);
>     my_loader_text.unref();
> }
> ```
> 
> But also, the definition of the loader class itself is missing the GDCLASS macro... IE:
> `GDCLASS(MyLoaderText, ResourceFormatLoader)`
> 
> There may be other errors, but these are what I have found thusfar.

**Comment by 99alfie** [comment]
Using a recent checkout Godot custom build c3.2.dev.custom_build.093798c97 on Windows

I made some progress, at least got it to compile, json files show up but "to_string" is not working.

attached is the project with my corrections (including the ones from Darkheron)

this error occurs on godot start:

```
ERROR: Class MyJson already has a method to_string.
   At: core\class_db.cpp:1225
```

when running the example, using "to_string" prints: [MyJson:RID], the Object::to_string() function, which means it has not been overridden

The docs on Object mention to override "_to_string", I tried that but it was not working either.

Only other example of a custom resource is visual_script and it is hard to figure out from there.

I tried using to_string(bool *r_valid), but it is never called, seems like code to implement *instance is missing, getting a bit stuck there.

 It would be very interesting for our project to have json available as resource. Thanks already.

[my_json.zip](https://github.com/godotengine/godot-docs/files/3525997/my_json.zip)

**Comment by 99alfie** [comment]
... "to_string" is not meant to serialize anything, it is just clashing with to_string from Object. Just needs another function name. Error checking is corrected too.

I will do a PR when I find the time, meanwhile here is the working example:

[my_json.zip](https://github.com/godotengine/godot-docs/files/3540499/my_json.zip)

**Comment by 99alfie** [comment]
here is another proposal for the example. Using "json" not "myjson", code is now up-to-date with the current API. I also added a "content" property with a getter/setter and a ResourceFormatSaver. With that, the json content is now visible / editable in the Inspector and it can be saved.
[json.zip](https://github.com/godotengine/godot-docs/files/3566350/json.zip)

**Comment by hungrymonkey** [comment]
Of course, its wrong. Docs like code needs to be refactored too. Godot is a living engine. I wrote that example during the godot 3.0 days.

https://github.com/godotengine/godot/commit/e8fc6bfeb53801447de25c12ba7536f574d121f4

Upstream changes their code style every other point release.
